drivers/sensor/ens160/ens160_i2c.c

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright (c) 2024 Gustavo Silva
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT sciosense_ens160
+
+#include <zephyr/drivers/i2c.h>
+
+#include "ens160.h"
+
+LOG_MODULE_DECLARE(ENS160, CONFIG_SENSOR_LOG_LEVEL);
+
+#if DT_ANY_INST_ON_BUS_STATUS_OKAY(i2c)
+
+static int ens160_read_reg_i2c(const struct device *dev, uint8_t reg, uint8_t *val)
+{
+	const struct ens160_config *config = dev->config;
+
+	return i2c_reg_read_byte_dt(&config->i2c, reg, val);
+}
+
+static int ens160_read_data_i2c(const struct device *dev, uint8_t reg, uint8_t *data, size_t len)
+{
+	const struct ens160_config *config = dev->config;
+
+	return i2c_burst_read_dt(&config->i2c, reg, data, len);
+}
+
+static int ens160_write_reg_i2c(const struct device *dev, uint8_t reg, uint8_t val)
+{
+	const struct ens160_config *config = dev->config;
+
+	return i2c_reg_write_byte_dt(&config->i2c, reg, val);
+}
+
+static int ens160_write_data_i2c(const struct device *dev, uint8_t reg, uint8_t *data, size_t len)
+{
+	const struct ens160_config *config = dev->config;
+
+	return i2c_burst_write_dt(&config->i2c, reg, data, len);
```

Comment:
Unfortunately i2c_burst_write is problematic, so should be avoided (https://github.com/zephyrproject-rtos/zephyr/issues/20154).

You will need to create a combined write buffer here, something like:

```suggestion
	__ASSERT(len == 2, "Only 2 byte write are supported");
	
	uint8_t buff[] = {reg, data[0], data[1]};

	return i2c_write_dt(&config->i2c, buff, sizeof(buff));
}
```
Alternatively you may wish to have a write 2 registers function instead.
If it is unlikely that there will be need to writes more than 2 bytes, this is a good option.

Reply:
I was not aware of this problem. Thank you for pointing it out.

The datasheet  (version 1.3) mentions in section 16.2.4 a write operation to a set of consecutive registers for configuring threshold interrupt, but some details are absent in the current revision. Therefore, this driver does not implement threshold triggers for the same reason.

However, I imagine that in a future revision the datasheet may become clearer and there will be the need for writing more than two bytes.

Because of this, I opted for your first suggestion.